bin/proving-service/build.rs

```diff
@@ -0,0 +1,68 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+};
+
+use miette::IntoDiagnostic;
+use protox::prost::Message;
+
+/// Defines whether the build script can write to /src.
+const CAN_WRITE_TO_SRC: bool = option_env!("DOCS_RS").is_none();
```

Comment:
I think we'll need to set this to true for publishing to crates.io as well. Maybe we should have a single environment variable that we set manually for this. This variable would be set when we run things via Makefile but otherwise, these type of scripts would be ignored.

Reply:
Should't we use the generated files that are committed in that case? Instead of re-building it?

Reply:
> I think we'll need to set this to true for publishing to crates.io as well
Wouldn't `DOCS_RS` not be set in that case and so the variable would be true?